plugin.video.eitb/video.py

```diff
@@ -0,0 +1,280 @@
+from utils import get_programs
+from utils import get_episodes
+from utils import get_playlists
+from utils import get_videos
+from utils import get_last_broadcast
+from utils import get_programs_types
+from utils import get_programs_types_playlist
+from utils import get_local_string
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import xbmc
+from constants import MENU_ITEMS
+
+addon = xbmcaddon.Addon()
+
+
+class VideoHandler(object):
+    def __init__(self, handle, url):
+        self.handle = handle
+        self.url = url
+
+    def list_menu(self):
+
+        listing = []
+        for item in MENU_ITEMS:
+            title_id = int(item.get("title_id"))
+            title = get_local_string(title_id)
+            option = item.get("@id")
+            icon = item.get("icon")
+            # Create a list item with a text label and a thumbnail image.
+            list_item = xbmcgui.ListItem(label=title)
+            list_item.setInfo("video", {"title": title})
+            list_item.setArt({"icon": addon.getAddonInfo("path") + icon})
+            url = "{0}?action=videomenu&option={1}".format(self.url, option)
+            is_folder = True
+            listing.append((url, list_item, is_folder))
+
+        # Set Content
+        xbmcplugin.setContent(self.handle, "tvshows")
+        # Add our listing to Kodi.
+        xbmcplugin.addDirectoryItems(self.handle, listing, len(listing))
+        # Add a sort method for the virtual folder items
+        xbmcplugin.addSortMethod(self.handle, xbmcplugin.SORT_METHOD_LABEL)
+        xbmcplugin.endOfDirectory(self.handle)
+
+    def list_programs(self):
+        """
+        Create the list of video programs in the Kodi interface.
+        """
+        programs = get_programs()
+        listing = []
+        for program in programs:
+            title = program.get("title")
+            program_url = program.get("@id")
+            # Create a list item with a text label and a thumbnail image.
+            list_item = xbmcgui.ListItem(label=title)
```

Comment:
you probably want to set `offscreen=True` on all your listitems

Reply:
done